apps/server/src/stores/EventStore.ts

```diff
@@ -7,6 +7,8 @@ export type StoreGetter = <T extends keyof RuntimeStore>(key: T) => Partial<Runt
 
 let store: Partial<RuntimeStore> = {};
 
+const changedKeys = new Array<keyof RuntimeStore>();
+let isUpdatePending: NodeJS.Immediate | null = null;
```

Comment:
this is an interesting way of batching updates in one event loop. nice!

Did you find this somewhere? I wonder whether `setImmediate` is more accurate to what we want instead of `setTimeout` or `nextTick`. Did you consider the alternatives?

I am especially wondering how this goes together with the scheduling of the store updates

Reply:
I tried reading through this

https://nodejs.org/en/learn/asynchronous-work/understanding-setimmediate

I'm not entirely sure what we actually want

I believe `setImmediate` runs at the end of the current event loop <---(I think this is a fitting spot)
and `setTimeout` may run on the start of the next loop
`nextTick` is the next tick in this current event loop <---(sounds like it would fire a bit to early)

Reply:
sounds good